File: c/ad-hoc/1091.c
/* title: Divisão da Nlogônia
 * problem number: 1091
 * date: 16/09/20
 * author: ltsdw
 */

#include <stdio.h>

char *return_cord(int cord_n, int cord_m, int x, int y);

int main(void) {
    int n_cases;
    while (scanf("%d", &n_cases), n_cases) {
        int cord_n, cord_m;
        scanf("%d%d", &cord_n, &cord_m);

        for (int i = 0; i < n_cases; i++) {
            int x, y;
            scanf("%d%d", &x, &y);
            printf("%s\n", return_cord(cord_n, cord_m, x, y));
        }
    }

    return 0;
}

char *return_cord(int cord_n, int cord_m, int x, int y) {
    if ( (x == cord_n) || (x == cord_m) || (y == cord_n) || (y == cord_m) ) {
        return "divisa";
    } else {
        if ( (x > cord_n) && (y > cord_m) ) {
            return "NE";
        } else if ( (x < cord_n) && (y > cord_m) ) {
            return "NO";
        } else if ( (x < cord_n) && (y < cord_m) ) {
            return "SO";
        } else {
            return "SE";
        }
    }
}

File: python/ad-hoc/1091.py
'''
title: Divisão da Nlogônia
number of the problem: 1091
date: 03/05/2018
author: ltsdw
'''

while(True):
    n_casos = int(input())
    if n_casos == 0: break
    else:
        cord_n, cord_m = map(int, input().split())
        for i in range(n_casos):
            x, y = map(int, input().split())
            if x == cord_n or x == cord_m or y == cord_n or y == cord_m:
                print('divisa')
            else:
                if x > cord_n and y > cord_m: print('NE')
                elif x < cord_n and y > cord_m: print('NO')
                elif x < cord_n and y < cord_m: print('SO')
                else: print('SE')

